Add documentation on how to visualise Matplotlib plots on Kedro-Viz (#1820)
kedro-org/kedro: docs/source/tutorial/visualise_pipeline.md

This will visualise the pipeline visualisation saved as `my_shareable_pipeline.json`.

> ## Visualising charts in Kedro-Viz

Reviewer: ```suggestion
## Visualise charts in Kedro-Viz
```
I tend to agree that "Visualising" sounds better here, but our style guide discourages the use of gerunds, especially in headings. When Jo (technical writer) is back she will be happy I said this anyway 😀

> Kedro-Viz aims to help users communicate different aspects of their workflow through an interactive flowchart. With the Plotly and Matplotlib integrations, we take one step further in this direction to allow our users to share their data insights effectively while exploring the pipeline.

Reviewer: Outside the scope of your change I know, but I'm personally not a big fan of this paragraph. It feels too vague and marketing-ese to me. Feel free to change it as you please or leave it as it is depending on what you think.

Author: @rashidakanchwala any thoughts on how to restructure this?

Reviewer 2: No idea, i personally liked this because it excites the user (by technically advertising the feature) but i agree with Antony too.

i will let @yetudada comment.

Reviewer 3: I wouldn't remove it completely - but I would suggest a modification. We describe Kedro-Viz here and also at the top of the page so I would remove the first sentence because we don't need to do it twice. 

So it could be something like: "Kedro-Viz supports [Plotly](https://plotly.com/python/) and [Matplotlib](https://matplotlib.org/) integrations so that you can share your data insights while exploring your pipeline."

![](../meta/images/pipeline_visualisation_plotly_expand.png)

### Visualise Matplotlib charts in Kedro-Viz

[Matplotlib](https://matplotlib.org/) is a Python library for creating static, animated, and interactive visualisations. Integrating Matplotlib into Kedro-Viz allows you to output your charts as part of your pipeline visualisation.

```{note}
The MatplotlibWriter dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the Plotly charts seen above.
```

You can view Matplotlib charts in Kedro-Viz when you use the [Kedro MatplotLibWriter dataset](/kedro.extras.datasets.matplotlib.MatplotlibWriter). You must update the `src/requirements.txt` file in your Kedro project by adding the following dataset to enable Matplotlib for your project:

```
kedro[matplotlib.MatplotlibWriter]==0.18.2
```

> To use this dataset, configure your plot in your Kedro node. The below functions can be added to the `nodes.py` and `pipeline.py` files respectively.

Reviewer 4: ```suggestion
To use this dataset, configure your plot in your Kedro node. The below functions should be added to the `nodes.py` and `pipeline.py` files respectively.
```

```python
# nodes.py
import matplotlib.pyplot as plt

def create_confusion_matrix(companies: pd.DataFrame):
    actuals = [0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 1]
    predicted = [1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1]
    data = {"y_Actual": actuals, "y_Predicted": predicted}
    df = pd.DataFrame(data, columns=["y_Actual", "y_Predicted"])
    confusion_matrix = pd.crosstab(
        df["y_Actual"], df["y_Predicted"], rownames=["Actual"], colnames=["Predicted"]
    )
    sn.heatmap(confusion_matrix, annot=True)
    return plt

# pipeline.py
def create_pipeline(**kwargs) -> Pipeline:
    """This is a simple pipeline which generates a plot"""
    return pipeline(
        [
            node(
                func=create_confusion_matrix,
                inputs="companies",
                outputs="dummy_confusion_matrix",
            ),
        ]
    )
```

You must also specify the output type in the `catalog.yml` file, like below.

```yaml
reporting.dummy_confusion_matrix:
  type: matplotlib.MatplotlibWriter
  filepath: ${base_location}/08_reporting/dummy_confusion_matrix.png

>   versioned: true

Reviewer 4: It might be useful to add a sentence about why `versioned: true` has been set here.

Author: Good point! Added an explanation in [14d7b98](https://github.com/kedro-org/kedro/pull/1820/commits/14d7b98411fb5e60c849fc7e5e278a3bf370ec38).

![](../meta/images/pipeline_visualisation_plotly_expand.png)

### Visualise Matplotlib charts in Kedro-Viz

> [Matplotlib](https://matplotlib.org/) is a Python library for creating static, animated, and interactive visualizations. Integrating Matplotlib into Kedro-Viz allows you to output your charts as part of your pipeline visualisation.

Reviewer: ```suggestion
[Matplotlib](https://matplotlib.org/) is a Python library for creating static, animated, and interactive visualisations. Integrating Matplotlib into Kedro-Viz allows you to output your charts as part of your pipeline visualisation.
```

Reviewer 2: not sure I am thinking maybe remove animated, and interactive since Kedro-viz only supports static. I know we add it somewhere in the bottom but it still might confuse.

Reviewer 2: Maybe we can put the note on Kedro-viz supports only static charts right after this.

> You can view Matplotlib charts in Kedro-Viz when you use the [Kedro MatplotLibWriter dataset](https://kedro.readthedocs.io/en/latest/kedro.extras.datasets.matplotlib.MatplotlibWriter.html). You must update the `requirements.txt` file in your Kedro project by adding the following dataset to enable Matplotlib for your project:

Reviewer: ```suggestion
You can view Matplotlib charts in Kedro-Viz when you use the [Kedro MatplotLibWriter dataset](/kedro.extras.datasets.matplotlib.MatplotlibWriter). You must update the `src/requirements.txt` file in your Kedro project by adding the following dataset to enable Matplotlib for your project:
```
Relative links are always better in our docs 🙂

> `kedro[matplotlib.MatplotlibWriter]==0.18.2`

Reviewer: ````suggestion
```
kedro[matplotlib.MatplotlibWriter]==0.18.2
```
````

Check how the built documentation renders but I think this is how we normally do code in a block (i.e. not inline).

> >**Note:** The MatplotlibWriter dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the Plotly charts seen above.

Reviewer: ````suggestion
```{note}
The MatplotlibWriter dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the Plotly charts seen above.
```
````

The syntax you used works on GitHub markdown but not Sphinx docs. Given that our documentation is consumed on RTD, which uses the sphinx build, it's more important to get it looking nice there.

> <!-- Add example on how to add Matplotlib charts-->
> 
> To use this dataset, configure your plot in your Kedro node.
> 
> Below is an example of how to visualise plots on Kedro-Viz using `matplotlib.MatplotlibWriter`.
> 
> The below functions can be added to the `nodes.py` and `pipeline.py` files respectively.

Reviewer: ```suggestion
To use this dataset, configure your plot in your Kedro node. The below functions can be added to the `nodes.py` and `pipeline.py` files respectively.
```

```python
# nodes.py
import matplotlib.pyplot as plt

> def create_confusion_matrix(companies: pd.DataFrame):

Reviewer 2: ```suggestion
def create_confusion_matrix(companies: pd.DataFrame):
    actuals = [0,1,0,0,1,1,1,0,1,0,1]
    predicted = [1,1,0,1,0,1,0,0,0,1,1]
    data = {"y_Actual": actuals, "y_Predicted": predicted}
    df = pd.DataFrame(data, columns=["y_Actual", "y_Predicted"])
    confusion_matrix = pd.crosstab(
        df["y_Actual"], df["y_Predicted"], rownames=["Actual"], colnames=["Predicted"]
    )
    sn.heatmap(confusion_matrix, annot=True)
    return plt
```

Reviewer 2: we could remove the random calc from the docs -- that was just for demo website to generate random plots for every run so we can distinguish one run from another.
